pallets/amm/src/lib.rs

```diff
@@ -354,6 +355,7 @@ pub mod pallet {
                                     .quote_amount
                                     .checked_sub(quote_amount)
                                     .ok_or(ArithmeticError::Underflow)?;
+                                *pool_liquidity_amount = Some(*liquidity_amount);
```

Comment:
this is not necessary since you are inside `try_mutate`

Reply:
replied above

Reply:
same as above